### PR TITLE
Processing data

### PR DESCRIPTION
**Pull request by Mahamadou-Nouridine**
In this Pr, I brought changes according to the following requirements:

### Create models
-  Create model classes for all entities as shown in the ERD diagram
-  Set up associations between models.
    - Remember that `author_id` column in `posts` table should be the foreign_key for the `users` table.

### Use models to insert data
- Familiarize yourself with ['rails c' command](https://guides.rubyonrails.org/command_line.html#bin-rails-console).
- Open rails console for the project.
- Create at least one user by running the following code:
    ```ruby
       first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
       second_user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.')
    ```
- Create at least 4 posts written by one of the users you created by running the following code:
    ```ruby
       first_post = Post.create(author: first_user, title: 'Hello', text: 'This is my first post')
    ```
- Create at least 6 posts comments for one of the posts you created by running the following code:
    ```ruby
       Comment.create(post: first_post, author: second_user, text: 'Hi Tom!' )
    ```
- Use other [CRUD methods](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/active_record_crud.md) to find, update and delete entities.

### Create custom methods
the models should include some additional methods.
-  Users
    - A method that returns the 3 most recent posts for a given user.
-  Posts
    - A method that updates the posts counter for a user.
    - A method which returns the 5 most recent comments for a given post.
-  Comments
    - A method that updates the comments counter for a post.
-  Likes
    - A method that updates the likes counter for a post.
- Go to `rails c` and check if your methods are working.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: 'author_id', dependent: :destroy
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
```
Comment (julie-ify) [inline app/models/user.rb]: - [x] Kindly specify the foreign key for the comments and likes which is `author_id`.

File: app/models/comment.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Comment < ApplicationRecord
+  belongs_to :user
```
Comment (julie-ify) [inline app/models/comment.rb]: - [x] Comments should belong to an author, not user. Kindly fix that. You can use the `class_name` property to point Rails to the User model.

  ```
  belongs_to :author, class_name: 'User'
  ```

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class Like < ApplicationRecord
+  belongs_to :user
```
Comment (julie-ify) [inline app/models/like.rb]: - [x] Likes should belong to an author, not user. Kindly fix that. You can use the `class_name` property to point Rails to the User model.

  ```
  belongs_to :author, class_name: 'User'
  ```

File: app/models/like.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  belongs_to :post
+  after_create :update_counter
+  after_destroy :update_counter
+
+  def update_counter
+    post.comments_count = post.comments.count
+    post.save
+  end
```
Comment (julie-ify) [inline app/models/like.rb:12]: - [x] **[OPTIONAL]** As a suggestion, I think this custom method should be private.

File: app/models/comment.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  belongs_to :post
+  after_create :update_counter
+  after_destroy :update_counter
+
+  def update_counter
+    post.comments_count = post.comments.count
+    post.save
+  end
```
Comment (julie-ify) [inline app/models/comment.rb:12]: - [x] **[OPTIONAL]** As a suggestion, I think this custom method should be private.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Mahamadou-Nouridine,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ridwanediallo — APPROVED** [review]
# STATUS: APPROVED 🟢
## Hi @Mahamadou-Nouridine , 
Congratulations!GIF

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- Well done implementing the requested changes. ✔
- The milestone requirements have been met with success. 🎉 🎉 🎉 ✔

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Mahamadou-Nouridine** [comment]
Thank you all @julie-ify @ridwanediallo

**Pull request merged**